File: backend/src/main/java/com/delivery/store/entity/Store.java
package com.delivery.store.entity;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;

@Entity
@Table(name = "stores")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Store {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, nullable = false)
    private String name;

    private String password;

    @Enumerated(EnumType.STRING)
    private Role role;

    @Column(unique = true, nullable = false)
    private String email;

    private String address;

    @Column(precision = 19, scale = 2)
    private BigDecimal balance;

    @Column(precision = 19, scale = 2)
    private BigDecimal reservedBalance;

    @Column(precision = 19, scale = 2)
    private BigDecimal totalSpent;

    private Boolean whalePass;

    private Long passId;

}

File: frontend/src/interfaces/Store.ts
export enum StoreTypeEnum {
    STORE,
    ADMIN
}

export interface Store {
    "id": number,
    "name": string,
    "password": string,
    "role": StoreTypeEnum,
    "email": string,
    "address": string,
    "balance": number,
    "reservedBalance": number,
    "totalSpent": number,
    "whalePass": boolean,
    "passId": string | null
}